AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -171,6 +171,94 @@ final class StorageEngine: StorageEngineBehavior {
         storageAdapter.delete(modelType, withId: id, completion: wrappedCompletion)
     }
 
+    func delete<M: Model>(_ modelType: M.Type,
+                          predicate: QueryPredicate,
+                          completion: @escaping DataStoreCallback<[M]>) {
+        var queriedResult: DataStoreResult<[M]>?
+        var deletedResult: DataStoreResult<[M]>?
+
+        let queryCompletionBlock: DataStoreCallback<[M]> = { queryResult in
+            queriedResult = queryResult
+            if case .success = queryResult {
+                let deleteCompletionWrapper: DataStoreCallback<[M]> = { deleteResult in
+                    deletedResult = deleteResult
+                }
+                self.storageAdapter.delete(modelType, predicate: predicate, completion: deleteCompletionWrapper)
+            }
+        }
+
+        do {
+            try storageAdapter.transaction {
+                storageAdapter.query(modelType,
+                                     predicate: predicate,
+                                     additionalStatements: nil,
+                                     completion: queryCompletionBlock)
+            }
+        } catch {
+            completion(.failure(causedBy: error))
+            return
+        }
+
+        let transactionResult = collapseResults(queryResult: queriedResult, deleteResult: deletedResult)
+
+        guard modelType.schema.isSyncable, let syncEngine = self.syncEngine else {
+            completion(transactionResult)
```

Comment:
maybe some debug logging in cases like this to help customers debug unexpected behavior? It could help us in case customers report the typical "my models are not syncing to the cloud"

Reply:
Done.. added to this function as well as the existing `delete()` function.